plugins/notifications-backend/src/service/router.ts

```diff
@@ -1,61 +1,160 @@
 import { errorHandler } from '@backstage/backend-common';
+import { CatalogClient } from '@backstage/catalog-client';
 
 import express from 'express';
 import Router from 'express-promise-router';
-import { Knex } from 'knex';
+import knex from 'knex';
 import { Logger } from 'winston';
+import { Config } from '@backstage/config';
 
 export interface RouterOptions {
   logger: Logger;
-  dbClient?: Knex<any, any[]>;
+  dbConfig: Config;
+  catalogClient: CatalogClient;
 }
 
+export type CreateRequest = {
+  subject: string;
+  body?: string;
+  targetUsers?: string[];
+  targetGroups?: string[];
+}
+
+
+
 export async function createRouter(
   options: RouterOptions,
 ): Promise<express.Router> {
-  const { logger, dbClient } = options;
+  const { logger, dbConfig, catalogClient } = options;
 
   const router = Router();
   router.use(express.json());
 
-  if (!dbClient) {
-    logger.error('Missing dbClient');
+  if (!dbConfig) {
+    logger.error('Missing dbConfig');
     return router;
   }
 
-  router.get('/health', (_, response) => {
-    logger.info('PONG!');
-    response.json({ status: 'ok' });
+  const dbClient = knex({
+    client: dbConfig.getString('client'),
+    connection: {
+      host: dbConfig.getString('connection.host'),
+      port: dbConfig.getNumber('connection.port'),
+      user: dbConfig.getString('connection.user'),
+      password: dbConfig.getString('connection.password'),
+      database: 'backstage_plugin_notifications'
+    }
+  });
+
+  if (! await dbClient.schema.hasTable('messages')) {
+    await dbClient.schema.createTable('messages', function (table) {
+      table.uuid("id", {primaryKey: true}).defaultTo(dbClient.fn.uuid());
+      table.string('subject');
+      table.string('body');
+    })
+
+  }
+
+  router.get('/health', async (_, response) => {
+    var myRes = 'ok';
+
+    const userRef = await catalogClient.getEntityByRef("user:jdoe");
+
+    if (!userRef) {
+      myRes = 'fail'
+    } else {
+      if (userRef.relations && userRef.relations.length > 0) {
+        userRef.relations.forEach((relation)=>{
+          const parts = relation.targetRef.split(':');
+          if (parts.length == 0 || parts[0] != 'group') {
+            return;
+          }
+          myRes = `${myRes}:${parts[1]}`;
+        });
+      }
+    }
+    
+    response.json({ status: myRes });
   });
 
-  router.get('/notifications/count', async (_, response) => {
+  router.get('/notifications/count/:text?', async ({params: {text}}, response) => {
```

Comment:
For extensibility, better to use:
```suggestion
  router.get('/notifications/count', async (request, response) => {
  const { text } = request.query;
  // ...
```

Reply:
exactly what i was looking at earlier - how to read the query params.
I must say I don't like it.
Why? express does not allow me to specify which query params i expect.
also I think the query becomes too big.
I would add a body to the GET with an object: ListNotificationsRequest

Reply:
I am old-school here and agree that we should not use the `body` in a GET handler - https://stackoverflow.com/questions/978061/http-get-with-request-body